Source/LibationFileManager/Configuration.PersistentSettings.cs

```diff
@@ -3,336 +3,250 @@
-
-        [Description("Automatically run periodic scans in the background?")]
-        public bool AutoScan
-        {
-            get => persistentDictionary.GetNonString<bool>(nameof(AutoScan));
-            set
-            {
-                if (AutoScan != value)
-                {
-                    persistentDictionary.SetNonString(nameof(AutoScan), value);
-                    AutoScanChanged?.Invoke(null, null);
-                }
-            }
-        }
-
-        [Description("Auto download books? After scan, download new books in 'checked' accounts.")]
-        // poorly named setting. Should just be 'AutoDownload'. It is NOT episode specific
-        public bool AutoDownloadEpisodes
-        {
-            get => persistentDictionary.GetNonString<bool>(nameof(AutoDownloadEpisodes));
-            set => persistentDictionary.SetNonString(nameof(AutoDownloadEpisodes), value);
-        }
-
-        [Description("Save all podcast episodes in a series to the series parent folder?")]
-        public bool SavePodcastsToParentFolder
-        {
-            get => persistentDictionary.GetNonString<bool>(nameof(SavePodcastsToParentFolder));
-            set => persistentDictionary.SetNonString(nameof(SavePodcastsToParentFolder), value);
-        }
+	public partial class Configuration
+	{
+		// note: any potential file manager static ctors can't compensate if storage dir is changed at run time via settings. this is partly bad architecture. but the side effect is desirable. if changing LibationFiles location: restart app
+
+		// default setting and directory creation occur in class responsible for files.
+		// config class is only responsible for path. not responsible for setting defaults, dir validation, or dir creation
+		// exceptions: appsettings.json, LibationFiles dir, Settings.json
+
+		private PersistentDictionary persistentDictionary;
+
+		public T GetNonString<T>([CallerMemberName] string propertyName = "") => persistentDictionary.GetNonString<T>(propertyName);
+		public object GetObject([CallerMemberName] string propertyName = "") => persistentDictionary.GetObject(propertyName);
+		public string GetString([CallerMemberName] string propertyName = "") => persistentDictionary.GetString(propertyName);
+		public void SetNonString(object newValue, [CallerMemberName] string propertyName = "")
+		{
+			var existing = getExistingValue(propertyName);
+			if (existing?.Equals(newValue) is true) return;
+
+			OnPropertyChanging(propertyName, existing, newValue);
+			persistentDictionary.SetNonString(propertyName, newValue);
+			OnPropertyChanged(propertyName, newValue);
+		}
+
+		public void SetString(string newValue, [CallerMemberName] string propertyName = "")
+		{
+			var existing = getExistingValue(propertyName);
+			if (existing?.Equals(newValue) is true) return;
+
+			OnPropertyChanging(propertyName, existing, newValue);
+			persistentDictionary.SetString(propertyName, newValue);
+			OnPropertyChanged(propertyName, newValue);
+		}
```

Comment:
All setters go through these two methods which take care of checking for changes and then firing the events. 

`[CallerMemberName] string propertyName = ""` allows the caller to not pass its own name, simplifying all the setting's property definitions.

Reply:
Finally! I've been meaning to do something like `[CallerMemberName]` for ages. So much cleaner

While we're mucking in all this deep settings stuff, should default values go here then we can rip out all of that ad hoc add-setting-if-not-exists migration code?

Reply:
Good idea. Let me think if there's an elegant way to do that. If it's inelegant, I'd just assume leave it the way it is and keep the Configuration file more readable.

Reply:
Agreed. That can be added anytime so I won't let that hold up this PR

Reply:
@rmcrackan I put together what I think is an elegant solution for default values, but it results in a change that I'm not sure you'd be happy with.  Basically, settings won't be written to the Settings.json file unless they are different from the default. Personally I like it because it means that anything in the Settings.json file is not default (excepting basic stuff like Serilog and Books directory).

Thoughts?

Reply:
Hmm, interesting. Feels uncomfortable but not wrong per se. I could live with that